pkg/controller/controller.go

```diff
@@ -443,14 +451,20 @@ func (p *csiProvisioner) Delete(volume *v1.PersistentVolume) error {
 	// get secrets if StorageClass specifies it
 	storageClassName := volume.Spec.StorageClassName
 	if len(storageClassName) != 0 {
-		if storageClass, err := p.client.StorageV1().StorageClasses().Get(storageClassName, metav1.GetOptions{}); err == nil {
+		storageClassObject, exists, err := p.classes.GetByKey(storageClassName)
+		glog.Infof("%+v %+v", exists, err)
+		if err == nil && exists {
```

Comment:
What happens when `err != nil`?  Add code to handle situations when error occurs.

Reply:
Ok.